Userland/Libraries/LibAudio/WavWriter.h

```diff
@@ -11,42 +11,38 @@
 #include <AK/RefPtr.h>
 #include <AK/StringView.h>
 #include <LibAudio/Sample.h>
-#include <LibCore/DeprecatedFile.h>
+#include <LibCore/File.h>
 #include <LibCore/Forward.h>
+#include <LibCore/Object.h>
 
 namespace Audio {
 
-class WavWriter {
+class WavWriter final : public Core::Object {
     AK_MAKE_NONCOPYABLE(WavWriter);
     AK_MAKE_NONMOVABLE(WavWriter);
+    C_OBJECT(WavWriter);
 
 public:
-    WavWriter(StringView path, int sample_rate = 44100, u16 num_channels = 2, u16 bits_per_sample = 16);
+    static ErrorOr<NonnullRefPtr<WavWriter>> try_create_from_file(StringView path, int sample_rate = 44100, u16 num_channels = 2, u16 bits_per_sample = 16);
```

Comment:
`try_` is reserved for APIs that already have an infallible variant and need to be distinguished by name, _not_ as a general prefix for fallible functions